pyproject.toml

```diff
@@ -1,6 +1,6 @@
 [project]
 name = "cumulus-library"
-requires-python = ">= 3.9"
+requires-python = ">= 3.11"
```

Comment:
Let me start with this change and we can discuss rest after.

Things this will make more difficult:
- Running in BCH's E2 environment. 3.9 is the only Python3 available via biogrids in that environment. There may be an alternate way to get a different python3 but I don't know how friendly that is. It would need research. So this change would make cumulus-library un-runnable in E2. -- That is probably OK for now? I don't have an immediate need for that, but it's something to note.
- Running on Mike's laptop. I'm still using the LTS release of 22.04 (until this April when 24.04 comes out and I'll probably upgrade). Python 3.10 is the default python here. I can install python 3.11 and with some work, can probably figure out how to use it for cumulus-library, but that's investigation I can do. Not a blocker per se, but possibly annoying. If you don't really need 3.11 features, is it alright to drop this down to 3.10 until 24.04 comes out and I upgrade?

Reply:
Per an offline discussion, we're going to hold to 3.10 until at least the next Ubuntu release as a baseline for 'what's reasonable to ask of others' guideline.

Reply:
Thank you :pray: for letting me be lazy for a little longer